File: pattern-jwt-server/src/main/java/fm/pattern/jwt/server/service/ScopeService.java
package fm.pattern.jwt.server.service;

import java.util.List;

import fm.pattern.jwt.server.model.Scope;
import fm.pattern.microstructure.Result;

public interface ScopeService {

	Result<Scope> create(Scope scope);

	Result<Scope> update(Scope scope);

	Result<Scope> delete(Scope scope);

	Result<Scope> findById(String id);

	Result<List<Scope>> list();

}

File: tokamak-console/app/src/services/ScopeService.js
import RestService from "./RestService.js"

var ScopeService = {
		
    list() {
        return RestService.list("/v1/scopes");
    },  
    
    create(name, description) {
        return RestService.post("/v1/scopes", { name:name, description:description });
    },

    update(id, name, description) {
        return RestService.put("/v1/scopes/" + id, { name:name, description:description });
    },

    delete(scope) {
        return RestService.delete("/v1/scopes/" + scope.id, scope);
    }     
    
}

export default ScopeService;